src/SDK.ts

```diff
@@ -709,14 +727,35 @@ export default class SDK implements ISDK {
       try {
         const response = await axiosInstance(options);
         const elapsedTimeInMilliseconds = timer.milliSecondsElapsed;
-        const { headers } = response;
+        const { data, headers } = response;
         this.setAuthCodeNonce(headers);
 
+        if (headers) {
+          if (headers[OmnichannelHTTPHeaders.ocSessionId.toLowerCase()]) {
+            this.sessionId = headers[OmnichannelHTTPHeaders.ocSessionId.toLowerCase()];
+          }
+        }
+
+        // Resolves only if it contains chat token response which only happens on status 200
+        if (data && isV2) {
+          
+          // check if data is empty, if so, then reject the promise
+          if (Object.keys(data).length === 0) {
+            reject(new Error("Empty data received from getChatToken"));
+            return;
+          }
+
+          data.requestId = requestId;
+          this.logWithLogger(LogLevel.INFO, OCSDKTelemetryEvent.SESSIONINITV2SUCCEEDED, "Session Init v2 Succeeded", requestId, response, elapsedTimeInMilliseconds, requestPath, method, undefined, undefined, requestHeaders);
+          resolve(data);
+          return;
+        }
+
```

Comment:
can we abstract the specific logic for v2 to a separate function, to reduce confusion and enhance code readability